sliceutil/sliceutil.go

```diff
@@ -160,6 +160,29 @@ func FilterString(list []string, fun func(string) bool) []string {
 	return ret
 }
 
+// RemoveStringUnordered removes a string from a list with no guarantee of order
+// returns the same list if the string is not found within it
+func RemoveStringUnordered(list []string, s string) []string {
```

Comment:
This only removes the first instance, whereas it would be easy to assume from the func name that it should remove any instances of it, what if there are duplicates and you want to remove all of them (but not remove any other duplicates so can't use UniqString)? 🤔

Reply:
Maybe we could do something like:
```go
var i int
for _, item := range list {
  if item != s {
    list[i] = item
    i++
  }
}
list = list[:i]
```

Reply:
damn it I wanted to do it without doing too much work around